src/subdomains/generic/user/models/auth/auth-lnurl.service.ts

```
@@ -0,0 +1,143 @@
+      authCacheEntry.accessToken = await this.signIn(signupDto, userIp);
+      authCacheEntry.accessTokenCreationTime = Date.now();
+
+      return AuthLnurlResponseDto.createOk();
+    } catch (e) {
+      this.logger.error('Failed to login with LNURL auth:', e);
+      return { status: AuthLnurlResponseStatus.ERROR, reason: 'invalid signature' };
+    }
+  }
+
+  private checkSignupDto(signupDto: AuthLnurlSignupDto): AuthLnurlResponseDto | undefined {
+    if ('login' !== signupDto.tag) return AuthLnurlResponseDto.createError('tag not found');
+    if ('login' !== signupDto.action) return AuthLnurlResponseDto.createError('action not found');
+
+    const k1 = signupDto.k1;
+    if (!k1) return AuthLnurlResponseDto.createError('challenge not found');
+
+    const signature = signupDto.sig;
+    if (!signature) return AuthLnurlResponseDto.createError('auth signature not found');
+
+    const key = signupDto.key;
+    if (!key) return AuthLnurlResponseDto.createError('key not found');
+
+    const authCacheEntry = this.authCache.get(k1);
+    if (!authCacheEntry) return AuthLnurlResponseDto.createError('challenge invalid');
+
+    const checkBeforeTime = Util.minutesBefore(5).getTime();
+    if (authCacheEntry.k1CreationTime < checkBeforeTime) return AuthLnurlResponseDto.createError('challenge expired');
+  }
+
+  async signIn(signupDto: AuthLnurlSignupDto, userIp: string): Promise<string> {
+    signupDto.wallet = 'DFX Bitcoin';
+
+    const session = { address: signupDto.address, signature: signupDto.signature };
+
+    const { accessToken } = await this.authService.signIn(session, true).catch((e) => {
+      if (e instanceof NotFoundException) return this.authService.signUp({ ...signupDto, ...session }, userIp, true);
+      throw e;
+    });
+
+    return accessToken;
+  }
+
+  getStatus(k1: string, signature: string, key: string): string {
+    const authCacheEntry = this.authCache.get(k1);
+    if (!authCacheEntry) return '';

```

I would throw a `NotFoundException`, if the k1 is not found.